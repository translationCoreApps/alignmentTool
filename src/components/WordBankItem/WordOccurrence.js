import React from 'react';
import PropTypes from 'prop-types';

const styles = {
  top: 0,
  marginTop: '2px',
  opacity: '0.8'
};

/**
 * Renders a words occurrence.
 * This is rendered as a superscript.
 * @param {int} occurrence - the order in which this word occurs (1 indexed).
 * @param {int} occurrences - how many times this word occurs in the context
 * @param {object} [style] - overrides the default styles
 * @return {*}
 * @constructor
 */
const WordOccurrence = ({occurrence, occurrences, style}) => {
  const computedStyles = {
    ...styles,
    ...style
  };
  if(occurrences > 1) {
    return <sup style={computedStyles}>{occurrence}</sup>;
  } else {
    return null;
  }
};

WordOccurrence.propTypes = {
  occurrence: PropTypes.int.isRequired,
  occurrences: PropTypes.int.isRequired,
  style: PropTypes.object
};

export default  WordOccurrence;
