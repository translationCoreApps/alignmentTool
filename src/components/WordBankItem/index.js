import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { DragSource } from 'react-dnd';
import ItemTypes from '../ItemTypes';

const internalStyle = {
  borderLeft: '5px solid #44C6FF',
  padding: '5px',
  margin: '10px',
  backgroundColor: '#FFFFFF',
  boxShadow: "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
  cursor: 'move'
};

const DragWordBankItemAction = {
  beginDrag(props) {
    // Return the data describing the dragged item
    const item = {
      word: props.word,
      occurrence: props.occurrence,
      occurrences: props.occurrences
    };
    return item;
  },
  endDrag(props, monitor, component) {
    // When dropped on a compatible target, do something
    // const item = monitor.getItem();
    // console.log(item)
  }
};

class WordBankItem extends Component {
  render() {
    const { word, isDragging, connectDragSource } = this.props;
    const opacity = isDragging ? 0.4 : 1;

    return connectDragSource(
      <div style={{ ...internalStyle, opacity }}>
        {word}
      </div>
    );
  }
}

WordBankItem.propTypes = {
  word: PropTypes.string.isRequired,
  occurrence: PropTypes.number.isRequired,
  occurrences: PropTypes.number.isRequired,
  connectDragSource: PropTypes.func.isRequired,
  isDragging: PropTypes.bool.isRequired
}

export default DragSource(
  ItemTypes.BOTTOM_WORD,
  DragWordBankItemAction,
  (connect, monitor) => ({
    connectDragSource: connect.dragSource(),
    isDragging: monitor.isDragging()
  })
)(WordBankItem);
