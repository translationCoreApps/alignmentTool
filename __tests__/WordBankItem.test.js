import React from 'react';
import WordBankItem from '../src/components/WordBankItem';
import renderer from 'react-test-renderer';

test('WordBankItem renders', () => {
    const component = renderer.create(
        <WordBankItem />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // TODO: exercise UI
});