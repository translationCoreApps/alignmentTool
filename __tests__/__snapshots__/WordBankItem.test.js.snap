// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordBankItem renders 1`] = `<DragSource(WordBankItem) />`;
